monix-execution/jvm/src/test/scala/monix/execution/misc/LocalJVMSuite.scala

```diff
@@ -94,6 +94,22 @@ object LocalJVMSuite extends SimpleTestSuite {
     for (v <- f) yield assertEquals(v, 50)
   }
 
+    testAsync("Local.bindCurrentIf(CancelableFuture) should properly keep current context during async boundaries on isolate=false") {
```

Comment:
I think this is a bad test.
- It's important that we get the use-case working (`Task.runToFuture.flatMap { ... future that reads local }`), not the internal package-private method
- Shouldn't be using single thread scheduler. If it does not work on multiple threads, then the use-case is broken which is something better seen in tests.

Reply:
Sure thing, do you have a recommendation of where to place such a test? Also should I keep this package private test as well (I personally think small isolated unit test is a good idea just so people don't accidentally break it in the future)

Reply:
LocalJVMSuite is good enough, since it's testing both JVM-specific stuff (threading) and Local interop.

Reply:
Thanks, will do this tonight

Reply:
I ended up adding a test to `TaskLocalJVMSuite` since you don't have access to `Task` in `LocalJVMSuite`. Let me know if you need anything else.